lib/awful/spawn.lua

```diff
@@ -6,6 +6,24 @@
 -- program after it has been launched.  This requires currently that the
 -- applicaton supports them.
 --
+-- **Rules of thumb when a shell is needed**:
+--
+-- * A shell is required when the commands contains `&&`, `;`, `||`, `&` or
+--  any other unix shell language syntax
+-- * When shell variables are defined as part of the command
+-- * When the command is a shell alias
+--
+-- Note there a shell is **not** a terminal emulator. A terminal emulator is
+-- something like XTerm, Gnome-terminal or Konsole. A shell is something like
+-- `bash`, `ZSH`, `busybox sh` or `Debian ash`.
+--
+-- If you wish to open a process in a terminal window, check that yours support
+-- the common `-e` option. If it does, then something like this should work:
+--
+--    awful.spawn(terminal.." -e 'my command'")
```

Comment:
Sadly, this is a problem.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=600429

With urxvt, you need the quotes. With gnome-terminal and terminator the quotes may not be present. Thus, you cannot portably execute a command containing spaces in a terminal.

I think this means we cannot describe this in the docs.